sql/mysql_db/mysql_db.go

```diff
@@ -767,11 +768,11 @@ func (db *MySQLDb) AuthMethod(user, addr string) (string, error) {
 	if addr == "@" || addr == "" {
 		host = "localhost"
 	} else {
-		splitHost, _, err := net.SplitHostPort(addr)
+		addrUrl, err := url.Parse(addr)
```

Comment:
Not sure if the semantics here are actually equivalent, or if we have sufficient test coverage to guarantee it

@Hydrocharged care to comment?

Reply:
Ah, the failing tests are related to this change. The reason I tried to swap is because `net.SplitHostPort` errors on port-less addresses. I don't think I can actually use `url.Parse` here as that function errors on IPs 
```
127.0.0.1:42626 // error
foobar:42626 // pass
```
which is news to me 😅. I will investigate a better way to handle port-less addresses

Reply:
Opted to keep the original `net.SplitHostPort` logic, and just added an exception for the port-less url error (which should be ok since this code only cares about extracting the host).

Reply:
Regarding the changes on what I was pinged for, I think the error-handling should be fine for missing ports. We do have _some_ test coverage of this path within Dolt, so I'd like to see a preliminary Dolt PR that references this commit as well. Assuming both have all passing tests, then I think it's fine to merge in.

Reply:
Thanks for reviewing. I've opened a preliminary PR against dolt with this branch of go-mysql-server https://github.com/dolthub/dolt/pull/7478

Reply:
Thanks for kicking off the CI on the dolt PR @Hydrocharged. Did you see the tests you were expecting to see pass? I noticed a couple failures that seem unrelated to the code change, but I'm not sure if they are preventing more of the CI from running. Failures like this related to pulling the branch: https://github.com/dolthub/dolt/actions/runs/7845504357/job/21471852227?pr=7478

I just brought both branches in sync with their respective `main`s, and fixed the go mods, just to be safe. Do you have any recommendations on how to proceed?